jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/FlowAlgorithm.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * (C) Copyright 2018-2018, by Joris Kinable and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.interfaces;
+
+import java.util.*;
+
+/**
+ * Iterface for flow algorithms
+ * @author Joris Kinable
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ */
+public interface FlowAlgorithm<V,E> {
+
+    /**
+     * Returns a <em>read-only</em> mapping from edges to the corresponding flow values.
+     *
+     * @return a <em>read-only</em> mapping from edges to the corresponding flow values.
+     */
+    Map<E, Double> getFlowMap();
+
+    /**
+     * For the specified {@code edge} $(u, v)$ return vertex $v$ if the flow goes from $u$ to $v$, or returns
+     * vertex $u$ otherwise. For directed flow networks the result is always the head of the specified arc.
+     * <p>
+     * <em>Note:</em> not all minimum cost flow algorithms may support undirected graphs.
+     *
+     * @param edge an edge from the specified flow network
+     * @return the direction of the flow on the {@code edge}
+     */
+    V getFlowDirection(E edge);
+
+    /**
+     * Represents a minimum cost flow.
+     *
+     * @param <E> graph edge type
+     * @since July 2018
```

Comment:
Do we still add `@since` tags to the classes?

Reply:
Nah we don't, they got removed since most of the classes didn't have them.